tower-http/src/classify.rs

```diff
@@ -0,0 +1,170 @@
+//! Tools for classifying responses as either success or failure.
+
+use http::{HeaderMap, Response, StatusCode};
+use http_body::Body;
+use std::{convert::Infallible, marker::PhantomData};
+
+/// Trait for classifying responses as either success or failure.
+pub trait ClassifyResponse {
+    /// The output of doing the classification.
+    ///
+    /// This could be an HTTP status code or some other kind of status.
+    type Output;
```

Comment:
wondering if this name could be a little more descriptive...maybe `Class`? but, that seems kind of ambiguous with the `Classification` type... hmm...

Reply:
I went with `FailureClass` in my next iteration because the success case no longer as any data, so this type is only included in the failure case.